### ISSUE TITLE
Converting item-list prediction to original item-id in the ETL work.

### ISSUE THREAD

**Issue by swapnilpanda** [NVIDIA-Merlin/Transformers4Rec#705]
### Discussed in https://github.com/NVIDIA-Merlin/Transformers4Rec/discussions/704

<div type='discussions-op-text'>

<sup>Originally posted by **swapnilpanda** May 18, 2023</sup>
I have been going through the example end-to-end prediction on transformer4rec, in the ETL notebook the item-id was shown in 6 digits, 
![image](https://github.com/NVIDIA-Merlin/Transformers4Rec/assets/56393206/94bbfed7-53c0-4f57-9a66-8ffa7d46b46c)

**however in the inference part, the item-id list we got had the item-id in 1 to 4 digit.**

![image](https://github.com/NVIDIA-Merlin/Transformers4Rec/assets/56393206/68f4581c-a891-41cf-bff7-8d48aac4a9d1)
 How do I track back to the original item-id so that I can actually make the inference and check my results? 
In other words, how is the product id in original data linked to item-id in the prediction, how are they mapped?
Any help will be appreciated, thanks. </div>

**Comment by swapnilpanda** [comment]
Thanks for the quick response. So if I understand you correctly this code should give me the correct product_ids

df = pd.read_parquet("/kaggle/input/etl-try1/categories/unique.prev_items.parquet")
df1 = df['prev_items']
decoded = []

for next_item in top_preds:
    decoded.append([df1.iloc[e-1] for e in next_item])

decoded = np.array(decoded)
As the given item_id is +1 of the original parquet file.

**Comment by NamartaVij** [comment]
Hi swapnil, have you completed this implementation. I have some doubts. Could you please help

**Comment by swapnilpanda** [comment]
Yes I have done this implementation, the function I am using is
        def _decoder(self, recommendation):
            
            '''decode sequeces to ASIN ids'''
            
            decoded = []
            for next_item in recommendation:

                decoded.append([self.items.iloc[e-1] for e in next_item])

            decoded = np.array(decoded)
            return decoded
where items is the unique.prev_items.parquet'

**Comment by NamartaVij** [comment]
Hi 
may I know how to do feature aggregation, my dataset is movielens , and my inputs are userid, movieid and genre and label is rating. So may I know how to do element wise concat. It would be great if you can help me in this, as I am getting error : might be this is the issue.   I followed this approach to creating the schema, but when i applied xlnet . i got an error https://nvidia-merlin.github.io/Merlin/stable/examples/getting-started-movielens/01-Download-Convert.html

**Comment by vivpra89** [comment]
@NamartaVij 

For decoding both input and output items: 

import pandas as pd

items = pd.read_parquet("./categories/unique.productId.parquet")['productId']
decoded_output = []
for next_item in prediction.predictions[0]:
    decoded_output.append([items.iloc[e-1] for e in next_item])

test = pd.read_parquet('./data/sessions_by_day/23/test.parquet')
decoded_input = []
for next_item in list(test['productId-list']):
    decoded_input.append([items.iloc[e-1] for e in next_item])

print(len(decoded_output),len(decoded_input))

predictions = pd.DataFrame({'input': decoded_input, 'prediction': decoded_output})
